packages/frontend-shared/src/gql-components/RecordRunModal.cy.tsx

```diff
@@ -0,0 +1,40 @@
+import RecordRunModalVue from './RecordRunModal.vue'
+
+describe('RecordRunModal', () => {
+  it('is not open by default', () => {
+    cy.mount(<RecordRunModalVue isModalOpen={false} close={cy.stub()} utmMedium="Nav" />)
+
+    cy.findByTestId('record-run-modal').should('not.exist')
+  })
+
+  it('renders open', () => {
+    cy.mount(<RecordRunModalVue isModalOpen={true} close={cy.stub()} utmMedium="Nav" />)
+
+    cy.contains('Record your first run').should('be.visible')
+    cy.findByTestId('copy-button').should('be.visible')
+
+    cy.percySnapshot()
+  })
+
+  it('calls close when X is clicked', () => {
+    const closeStub = cy.stub()
+
+    cy.mount(<RecordRunModalVue isModalOpen={true} close={closeStub} utmMedium="Nav" />)
+
+    cy.findByRole('button', { name: 'Close' }).click().then(() => {
+      expect(closeStub).to.have.been.called
+    })
+  })
+
+  it('sends UTM parameters with help link', () => {
+    cy.mount(<RecordRunModalVue isModalOpen={true} close={cy.stub()} utmMedium="Nav" utmContent="content" />)
+
+    cy.contains('Need help').should('have.attr', 'href', 'https://on.cypress.io/cypress-run-record-key?utm_medium=Nav&utm_source=Binary%3A+Launchpad&utm_content=content')
+  })
+
+  it('sends UTM parameters with help link without UTM content prop', () => {
+    cy.mount(<RecordRunModalVue isModalOpen={true} close={cy.stub()} utmMedium="Nav" />)
+
+    cy.contains('Need help').should('have.attr', 'href', 'https://on.cypress.io/cypress-run-record-key?utm_medium=Nav&utm_source=Binary%3A+Launchpad&utm_content=')
+  })
```

Comment:
Can we add a test case for if the gql query fails? Seems like we may end up with an empty/invalid record command if we fail to retrieve record keys for whatever reason

Reply:
I do have a test in `RecordPrompt.cy.tsx` that I think covers this case. If we fail to retrieve a record key then we just put `<record-key>` in the command
```
  it('renders with no props', () => {
    cy.mount(<RecordPromptVue />)

    cy.findByText('npx cypress run --record --key <record-key>').should('be.visible')
  })
```